tools/run_tests/build_artifact_python.sh

```diff
@@ -81,5 +81,19 @@ then
   done
 fi
 
+# We need to use the built grpcio-tools/grpcio to compile the health proto
+# Wheels are not supported by setup_requires/dependency_links, so we
+# manually install the dependency
+if [ "$SKIP_PIP_INSTALL" == "" ]
```

Comment:
I think `$SKIP_PIP_INSTALL` is more an optimization-hint-thing; if we need to install the built `grpcio-tools` for correctness then we probably want to skip `$SKIP_PIP_INSTALL`.